Make check fails with maxset

Making check in config
make[1]: Entering directory `/auto.anoa/home/weeber/es-builds/maxset/config'
make[1]: Nothing to be done for `check'.
make[1]: Leaving directory `/auto.anoa/home/weeber/es-builds/maxset/config'
Making check in src
make[1]: Entering directory `/auto.anoa/home/weeber/es-builds/maxset/src'
Making check in core
make[2]: Entering directory `/auto.anoa/home/weeber/es-builds/maxset/src/core'
  GEN      myconfig-final.hpp <= ../../myconfig.hpp
make  check-recursive
make[3]: Entering directory `/auto.anoa/home/weeber/es-builds/maxset/src/core'
Making check in unit_tests
make[4]: Entering directory `/auto.anoa/home/weeber/es-builds/maxset/src/core/unit_tests'
make  Vector_test
make[5]: Entering directory `/auto.anoa/home/weeber/es-builds/maxset/src/core/unit_tests'
  CXX      Vector_test.o
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In function ‘bool il_constructor()’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:19: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                   ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In function ‘bool norm2()’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:19: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                   ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 1]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:51:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<1, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 1; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<1, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 1; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 1; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 2]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:52:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<2, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 2; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<2, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 2; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 2; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 3]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:53:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<3, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 3; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<3, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 3; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 3; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 4]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:54:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<4, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 4; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<4, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 4; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 4; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 5]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:55:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<5, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 5; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<5, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 5; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 5; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 6]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:56:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<6, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 6; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<6, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 6; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 6; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 7]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:57:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<7, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 7; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<7, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 7; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 7; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 8]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:58:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<8, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 8; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<8, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 8; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 8; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 9]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:59:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<9, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 9; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<9, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 9; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 9; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 10]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:60:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<10, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 10; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<10, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 10; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 10; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 1]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:75:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<1, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 1; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<1, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 1; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 1; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 2]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:76:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<2, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 2; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<2, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 2; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 2; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 3]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:77:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<3, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 3; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<3, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 3; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 3; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 4]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:78:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<4, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 4; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<4, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 4; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 4; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 5]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:79:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<5, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 5; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<5, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 5; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 5; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 6]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:80:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<6, int>::Vector(<brace-enclosed initializer list>)’
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 6; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<6, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 6; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 6; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
make[5]: *** [Vector_test.o] Error 1
make[5]: Leaving directory `/auto.anoa/home/weeber/es-builds/maxset/src/core/unit_tests'
make[4]: *** [check-am] Error 2
make[4]: Leaving directory `/auto.anoa/home/weeber/es-builds/maxset/src/core/unit_tests'
make[3]: *** [check-recursive] Error 1
make[3]: Leaving directory `/auto.anoa/home/weeber/es-builds/maxset/src/core'
make[2]: *** [check] Error 2
make[2]: Leaving directory `/auto.anoa/home/weeber/es-builds/maxset/src/core'
make[1]: *** [check-recursive] Error 1
make[1]: Leaving directory `/auto.anoa/home/weeber/es-builds/maxset/src'
make: *** [check-recursive] Error 1

Actually, this happens in a --without-cuda build.

Making check in config
make[1]: Entering directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/config'
make[1]: Nothing to be done for `check'.
make[1]: Leaving directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/config'
Making check in src
make[1]: Entering directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src'
Making check in core
make[2]: Entering directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core'
  GEN      myconfig-final.hpp <= ../../myconfig.hpp
make  check-recursive
make[3]: Entering directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core'
Making check in unit_tests
make[4]: Entering directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core/unit_tests'
make  Vector_test
make[5]: Entering directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core/unit_tests'
  CXX      Vector_test.o
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 1]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:51:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<1, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 1; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<1, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 1; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 1; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 2]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:52:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<2, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 2; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<2, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 2; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 2; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 3]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:53:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<3, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 3; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<3, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 3; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 3; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 4]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:54:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<4, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 4; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<4, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 4; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 4; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 5]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:55:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<5, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 5; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<5, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 5; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 5; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 6]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:56:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<6, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 6; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<6, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 6; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 6; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 7]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:57:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<7, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 7; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<7, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 7; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 7; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 8]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:58:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<8, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 8; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<8, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 8; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 8; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 9]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:59:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<9, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 9; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<9, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 9; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 9; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool il_constructor() [with int n = 10]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:60:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: error: no matching function for call to ‘Vector<10, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:18:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 10; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<10, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 10; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 10; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 1]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:75:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<1, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 1; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<1, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 1; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 1; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 2]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:76:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<2, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 2; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<2, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 2; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 2; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 3]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:77:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<3, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 3; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<3, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 3; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 3; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 4]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:78:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<4, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 4; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<4, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 4; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 4; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 5]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:79:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<5, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 5; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<5, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 5; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 5; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp: In instantiation of ‘bool norm2() [with int n = 6]’:
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:80:3:   required from here
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: error: no matching function for call to ‘Vector<6, int>::Vector(<brace-enclosed initializer list>)’
   Vector<n, int> v(TEST_NUMBERS);
                                ^
/home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:42:32: note: candidates are:
In file included from /home/weeber/espresso-dev/src/core/unit_tests/Vector_test.cpp:3:0:
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note: Vector<n, Scalar>::Vector(const Vector<n, Scalar>&) [with int n = 6; Scalar = int]
   Vector(const Vector& rhs) {
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:53:3: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘const Vector<6, int>&’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note: Vector<n, Scalar>::Vector(Scalar*) [with int n = 6; Scalar = int]
   explicit Vector(Scalar *a) {
            ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:32:12: note:   no known conversion for argument 1 from ‘<brace-enclosed initializer list>’ to ‘int*’
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note: Vector<n, Scalar>::Vector() [with int n = 6; Scalar = int]
   Vector() {};
   ^
/home/weeber/espresso-dev/src/core/unit_tests/../Vector.hpp:30:3: note:   candidate expects 0 arguments, 1 provided
make[5]: *** [Vector_test.o] Error 1
make[5]: Leaving directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core/unit_tests'
make[4]: *** [check-am] Error 2
make[4]: Leaving directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core/unit_tests'
make[3]: *** [check-recursive] Error 1
make[3]: Leaving directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core'
make[2]: *** [check] Error 2
make[2]: Leaving directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src/core'
make[1]: *** [check-recursive] Error 1
make[1]: Leaving directory `/auto.icpopene/work/weeber/2013-01-18-grow-large-gels/src'
make: *** [check-recursive] Error 1

--- a/doc/ug/part.tex
+++ b/doc/ug/part.tex
@@ -1070,10 +1070,10 @@ constraint  \opt{\var{num}}
 Prints out all constraint information. If \var{num} is specified only this
 constraint is displayed, otherwise all constraints will be printed.
 
-\subsection{\texttt{harmonic_force}: Creating a harmonic trap}
- \newescommand{harmonic-force}
+\subsection{\texttt{harmonic_well}: Creating a harmonic trap}
+ \newescommand{harmonic-well}
  \begin{essyntax}
-   harmonic_force \{ \var{x} \var{y} \var{z} \} \var{k}
+   harmonic_well \{ \var{x} \var{y} \var{z} \} \var{k}
   \begin{features}
     \required{CUDA}
   \end{features}

--- a/src/core/Vector.hpp
+++ b/src/core/Vector.hpp
@@ -22,17 +22,20 @@
 #include <initializer_list>
 #endif
 
+/* @TODO: should have move semantics with c++11 for peformance reasons. */
+
 template<int n, typename Scalar>
 class Vector {
-private:
+ private:
   Scalar d[n];
-public:
-  Vector() {};
+  
+ public:
+  Vector() {}
 
-  explicit Vector(Scalar *a) {
+  explicit Vector(const Scalar * const a) {
     for (int i = 0; i < n; i++)
       d[i] = a[i];
-  };
+  }
 
 #ifdef HAVE_CXX11
   Vector(std::initializer_list<Scalar> l) {
@@ -51,58 +54,64 @@ public:
   int size() const { return n; }
   
   Vector(const Vector& rhs) {
-    std::copy(rhs.d,rhs.d+n,d);
+    std::copy(rhs.d, rhs.d+n, d);
   }
 
   void swap(Vector& rhs) {
-    std::swap(d,rhs.d);
+    std::swap(d, rhs.d);
   }
 
   Vector& operator=(Vector& rhs) {
     Vector tmp(rhs); swap(rhs); return *this;
-  };
+  }
 
-  Scalar &operator[](int i) const {
+  Scalar &operator[](int i) {
     return d[i];
-  };
+  }
+
+  Scalar const & operator[](int i) const {
+    return d[i];
+  }
   
   inline Scalar dot(const Vector<n, Scalar> &b) const {
     Scalar sum = 0;
-    for(int i = 0; i < n; i++)
+    for (int i = 0; i < n; i++)
       sum += d[i]*b[i];
     return sum;
-  };
+  }
 
   inline Scalar norm2(void) const {
     return dot(*this);
-  };
+  }
 
   inline Scalar norm(void) const {
     return sqrt(norm2());
-  };
+  }
 
   inline void normalize(void) {
     Scalar N = norm();
-    if(norm() > 0) {
-      for(int i = 0; i < n; i++)
-	d[i] /= N;
+    if (norm() > 0) {
+      for (int i = 0; i < n; i++)
+        d[i] /= N;
     }
-  };
+  }
 
   inline void cross(const Vector<3, Scalar> &a, const Vector<3, Scalar> &b, Vector<3, Scalar> &c) const {
     c[0] = a[1]*b[2] - a[2]*b[1];
     c[1] = a[2]*b[0] - a[0]*b[2];
     c[2] = a[0]*b[1] - a[1]*b[0];
     return;
-  };
+  }
+  
   inline Vector<3, Scalar> cross(const Vector<3, Scalar> &a, const Vector<3, Scalar> &b) const {
     Vector<3, Scalar> c;
     cross(a,b,c);
     return c;
-  };
+  }
+  
   inline Vector<3, Scalar> cross(const Vector<3, Scalar> &a) const {
     return cross(this,a);
-  };
+  }
 };
 
 typedef Vector<3, double> Vector3d;

--- a/src/core/config.hpp
+++ b/src/core/config.hpp
@@ -32,6 +32,11 @@
 /* Include the defines created by configure. */
 #include <acconfig.hpp>
 
+/* If nvcc is used we disable c++11 support. */
+#if defined(__CUDACC__) && defined(HAVE_CXX11)
+#undef HAVE_CXX11
+#endif
+
 /* Prevent C++ bindings in MPI (there is a DataType called LB in there) */
 #define OMPI_SKIP_MPICXX
 #define MPICH_SKIP_MPICXX

--- a/src/core/unit_tests/Vector_test.cpp
+++ b/src/core/unit_tests/Vector_test.cpp
@@ -1,9 +1,34 @@
+/*
+  Copyright (C) 2010,2011,2012,2013,2014,2015 The ESPResSo project
+  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010 
+    Max-Planck-Institute for Polymer Research, Theory Group
+  
+  This file is part of ESPResSo.
+  
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
 
-#include <iostream>
-#include "../Vector.hpp"
+/** \file Vector_test.cpp Unit tests for the Utils::Vector class.
+ *
+*/
 
 #define BOOST_TEST_MODULE Vector test
 #include <boost/test/included/unit_test.hpp>
+
+#include <algorithm>
+
+#include "../Vector.hpp"
 
 /** Number of nontrivial Baxter permutations of length 2n-1. (A001185) */
 #define TEST_NUMBERS { 0, 1, 1, 7, 21, 112, 456, 2603, 13203 }
@@ -12,25 +37,27 @@ const int test_numbers[] = TEST_NUMBERS;
 const int test_numbers_partial_norm2[] = TEST_NUMBERS_PARTIAL_NORM2;
 const int n_test_numbers = sizeof(test_numbers) / sizeof(int);
 
+#ifdef HAVE_CXX11
 template <int n>
 bool il_constructor() {
   bool pass = true;
   Vector<n, int> v(TEST_NUMBERS);
 
-  pass &= v.size() == n; 
+  pass &= v.size() == n;
   
-  for(int i = 0; i < std::min(n_test_numbers, n); i++)
+  for (int i = 0; i < std::min(n_test_numbers, n); i++)
     pass &= v[i] == test_numbers[i];
     
   return pass;
 }
+#endif
 
 template<int n>
 bool default_constructor() {
   bool pass = true;
   Vector<n, int> v;
 
-  for(int i = 0; i < n; i++) {
+  for (int i = 0; i < n; i++) {
     v[i] = i;
   }
   
@@ -39,15 +66,13 @@ bool default_constructor() {
 
 template<int n>
 bool norm2() {
-  Vector<n, int> v(TEST_NUMBERS);
+  Vector<n, int> v(test_numbers);
 
-  std::cout << v.norm2() << " == " << test_numbers_partial_norm2[n] << std::endl;
-  
   return v.norm2() == test_numbers_partial_norm2[n-1];
 }
 
-BOOST_AUTO_TEST_CASE( test_constructor )
-{
+BOOST_AUTO_TEST_CASE(test_constructor) {
+#ifdef HAVE_CXX11
   BOOST_CHECK(il_constructor<1>());
   BOOST_CHECK(il_constructor<2>());
   BOOST_CHECK(il_constructor<3>());
@@ -58,6 +83,7 @@ BOOST_AUTO_TEST_CASE( test_constructor )
   BOOST_CHECK(il_constructor<8>());
   BOOST_CHECK(il_constructor<9>());
   BOOST_CHECK(il_constructor<10>());
+#endif
 
   BOOST_CHECK(default_constructor<1>());
   BOOST_CHECK(default_constructor<2>());
@@ -68,14 +94,14 @@ BOOST_AUTO_TEST_CASE( test_constructor )
   BOOST_CHECK(default_constructor<7>());
   BOOST_CHECK(default_constructor<8>());
   BOOST_CHECK(default_constructor<9>());
-  BOOST_CHECK(default_constructor<10>());  
+  BOOST_CHECK(default_constructor<10>());
 }
 
-BOOST_AUTO_TEST_CASE( test_norm2 ) {
+BOOST_AUTO_TEST_CASE(test_norm2) {
   BOOST_CHECK(norm2<1>());
   BOOST_CHECK(norm2<2>());
   BOOST_CHECK(norm2<3>());
   BOOST_CHECK(norm2<4>());
   BOOST_CHECK(norm2<5>());
-  BOOST_CHECK(norm2<6>()); 
+  BOOST_CHECK(norm2<6>());
 }

